src/type.h

```diff
@@ -158,4 +158,117 @@ class TypeSChar : public IntegralTypeHelper<signed char> {
     void dbgDump();
 };
 
+class TypeUChar : public IntegralTypeHelper<unsigned char> {
+  public:
+    TypeUChar(bool _is_static, CVQualifier _cv_qual)
+        : IntegralTypeHelper(_is_static, _cv_qual) {}
+
+    IntTypeID getIntTypeId() final { return IntTypeID::UCHAR; }
+    std::string getName() final { return "unsigned char"; }
+
+    void dbgDump();
+};
+
+class TypeSShort : public IntegralTypeHelper<short> {
+  public:
+    TypeSShort(bool _is_static, CVQualifier _cv_qual)
+        : IntegralTypeHelper(_is_static, _cv_qual) {}
+
+    IntTypeID getIntTypeId() final { return IntTypeID::SHORT; }
+    std::string getName() final { return "short"; }
+
+    void dbgDump();
+};
+
+class TypeUShort : public IntegralTypeHelper<unsigned short> {
+  public:
+    TypeUShort(bool _is_static, CVQualifier _cv_qual)
+        : IntegralTypeHelper(_is_static, _cv_qual) {}
+
+    IntTypeID getIntTypeId() final { return IntTypeID::USHORT; }
+    std::string getName() final { return "unsigned short"; }
+
+    void dbgDump();
+};
+
+class TypeSInt : public IntegralTypeHelper<int> {
+  public:
+    TypeSInt(bool _is_static, CVQualifier _cv_qual)
+        : IntegralTypeHelper(_is_static, _cv_qual) {}
+
+    IntTypeID getIntTypeId() final { return IntTypeID::INT; }
+    std::string getName() final { return "int"; }
+
+    void dbgDump();
+};
+
+class TypeUInt : public IntegralTypeHelper<unsigned int> {
+  public:
+    TypeUInt(bool _is_static, CVQualifier _cv_qual)
+        : IntegralTypeHelper(_is_static, _cv_qual) {}
+
+    IntTypeID getIntTypeId() final { return IntTypeID::UINT; }
+    std::string getName() final { return "unsigned int"; }
+    std::string getLiteralSuffix() final { return "U"; }
+
+    void dbgDump();
+};
+
+//  TODO: add bit mode to command-line options
+#if BIT_MODE64 == true
```

Comment:
It seems to be bad idea to have it as compile-time parameter (i.e. macro), rather than runtime value (which can be passed via command line to yarpgen). I think we need think how to make runtime parameter.

Reply:
"Long" type was removed. The idea that IR types are backend types was inforced.